cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -394,13 +396,124 @@ func deploymentNameForEnv(envName string, clock clock.Clock) string {
 	return name[len(name)-cArmDeploymentNameLengthMax:]
 }
 
+// prevDeploymentResult looks and finds a previous deployment for the current azd project.
+func (p *BicepProvider) prevDeploymentResult(
+	ctx context.Context,
+	scope infra.Scope,
+) (*armresources.DeploymentExtended, error) {
+	deployments, err := p.findCompletedDeployments(ctx, p.env.GetEnvName(), scope, "")
+	// findCompletedDeployments returns error if no deployments are found
+	// No need to check for empty list
+	if err != nil {
+		return nil, err
+	}
+
+	if len(deployments) > 1 {
+		// If more than one deployment found, ignore the prev-deployment
+		return nil, fmt.Errorf("more than one previous deployment match.")
+	}
+
+	return deployments[0], nil
+}
+
+// applyParameters returns parameters in the same form as returned by a previous deployment, where parameters values have
+// been applied to the template's parameters.
+func applyParameters(
+	templateParameters azure.ArmTemplateParameterDefinitions, params azure.ArmParameters) map[string]interface{} {
+	result := make(map[string]interface{}, len(templateParameters))
+	for paramName, paramDefinition := range templateParameters {
+		data := make(map[string]interface{}, 2)
+		data["type"] = paramDefinition.Type
+		data["value"] = paramDefinition.DefaultValue
+		if param, exists := params[paramName]; exists {
+			data["value"] = param.Value
+		}
+		result[paramName] = data
+	}
+	return result
+}
+
+// prevDeploymentEqualToCurrent compares the template hash from a previous deployment against a current template.
+// It also compares the set of parameters and values
+func (p *BicepProvider) prevDeploymentEqualToCurrent(
+	ctx context.Context, prev *armresources.DeploymentExtended, current *bicepDeploymentDetails) bool {
+	prevTemplateHash := convert.ToValueWithDefault(prev.Properties.TemplateHash, "")
+
+	createHashResult, err := p.deploymentsService.CalculateTemplateHash(
+		ctx, p.env.GetSubscriptionId(), current.CompiledBicep.RawArmTemplate)
+	if err != nil {
+		logADS("can't get hash from current template: %v", err)
+		return false
+	}
+	if prevTemplateHash != *createHashResult.TemplateHash {
+		logADS("template hash is different from previous deployment")
+		return false
+	}
+
+	currentParams := applyParameters(current.CompiledBicep.Template.Parameters, current.Parameters)
+	prevParams, goodCast := prev.Properties.Parameters.(map[string]interface{})
+	if !goodCast {
+		logADS("unexpected type found on previous deployment parameters. Ignoring deployment")
+		return false
+	}
```

Comment:
The creation of the template hash along with the comparison of the previous template make this method a bit hard to follow what's going on.  Consider just having separate method to create the hash and then another to perform the actual comparison.